server/config/config.go

```diff
@@ -305,6 +305,13 @@ type HTTPBasicAuthConfig struct {
 	Password string `json:"password" yaml:"password"`
 }
 
+// PackagingConfig holds configuration to build and retrieve Fleet packages
+type PackagingConfig struct {
```

Comment:
didn't find a good place to put this config so I created a new sub-category, in which we can add the `S3` configs too in a subsequent PR.

I've no problems changing this if something else makes sense.

Reply:
What about putting it under the top-level `FleetConfig`? Strictly speaking, you don't need to generate and install packages to enroll hosts with fleet. Some people will use plain osquery.

Reply:
That's a good place too, I can see how this can be used as a more general config.

Context: there are a bunch of S3 config items (similar to the ones for craves) coming up next for packaging, and that's part of the motivation of having a separate group for all packaging-related configs.

I feel like if we want to communicate that's OK to use this config outside of packaging we should move it, and vice-versa, we should keep it here if we want to discourage its usage.